aws-ssm-maintenancewindowtarget/aws-ssm-maintenancewindowtarget.json

```diff
@@ -0,0 +1,232 @@
+                    "maxLength": 128,
+                    "pattern": "^[a-zA-Z0-9_\\-.]{3,128}$"
+                },
+                "OwnerInformation": {
+                    "description": "A user-provided value that will be included in any CloudWatch events that are raised while running tasks for these targets in this maintenance window.",
+                    "type": "string",
+                    "minLength": 1,
+                    "maxLength": 128
+                },
+                "ResourceType": {
+                    "description": "The type of target that is being registered with the maintenance window.",
+                    "type": "string",
+                    "enum": [
+                        "INSTANCE",
+                        "RESOURCE_GROUP"
+                    ]
+                },
+                "Targets": {
+                    "description": "The targets, either instances or tags.",
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/Target"
+                    },
+                    "minItems": 0,
+                    "maxItems": 5
+                },
+                "WindowId": {
+                    "description": "The ID of the maintenance window to register the target with.",
+                    "type": "string",
+                    "minLength": 20,
+                    "maxLength": 20,
+                    "pattern": "^mw-[0-9a-f]{17}$"
+                },
+                "WindowTargetId": {
+                    "description": "The ID of the target.",
+                    "type": "string",
+                    "minLength": 36,
+                    "maxLength": 36,
+                    "pattern": "^[0-9a-fA-F]{8}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{4}\\-[0-9a-fA-F]{12}$"
+                }
+            }
+        }
+    },
+    "properties": {
+        "ClientToken": {
+            "description": "User-provided idempotency token.",
+            "type": "string",
+            "minLength": 1,
+            "maxLength": 64
+        },
+        "Description": {
+            "description": "A description for the target.",
+            "type": "string",
+            "minLength": 1,
+            "maxLength": 128
+        },
+        "Filters": {
+            "description": "Optional filters that can be used to narrow down the scope of the returned window targets. The supported filter keys are Type, WindowTargetId and OwnerInformation.",
+            "type": "array",
+            "items": {
+                "$ref": "#/definitions/MaintenanceWindowFilter"
+            },
+            "minItems": 0,
+            "maxItems": 5
+        },
+        "MaxResults": {
+            "description": "The maximum number of items to return for this call. The call also returns a token that you can specify in a subsequent call to get the next set of results.",
+            "type": "integer",
+            "minLength": 10,
+            "maxLength": 100
```

Comment:
Shouldn't the `integer` range constraints be `minimum` and `maximum` instead of `minLength` and `maxLength`?
https://json-schema.org/understanding-json-schema/reference/numeric.html#range

https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-kinesisfirehose/pull/16#issuecomment-663757793
https://github.com/aws-cloudformation/cloudformation-cli/issues/414

Reply:
Also, `describe` options like `Filters`, `MaxResults`, and `NextToken` should not be properties of the resource itself:
https://docs.aws.amazon.com/cli/latest/reference/ssm/describe-maintenance-window-targets.html
https://docs.aws.amazon.com/cli/latest/reference/ssm/update-maintenance-window-target.html

Reply:
Thanks, I've made the change for minimum and maximum in a different PR.

As for the describe option, I having trouble finding where in the documentation it specifies that `Filters`, `MaxResults`, and `NextToken` should not be properties. It seems like the inputs/outputs can include any of them: https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_DescribeMaintenanceWindowTargets.html.

Reply:
There will be API parameters for operations related to resources that aren't really properties of the resource itself. It doesn't make sense for something like `NextToken` to be specified in a CloudFormation template that provisions a resource

Reply:
I made the fix in this PR: https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-ssm/pull/75